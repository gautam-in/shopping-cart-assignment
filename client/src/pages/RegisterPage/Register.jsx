import React from "react";
import "./../LoginPage/Form.css";
import { useForm } from "react-hook-form";
import { useNavigate } from "react-router";
import { yupResolver } from "@hookform/resolvers/yup";
import * as Yup from "yup";
import Button from "../../components/Button/Button";

function Register() {
  const navigate = useNavigate();

  const validationSchema = Yup.object().shape({
    password: Yup.string()
      .required("Password is required")
      .min(6, "Password must be at least 6 characters."),
    confirmpassword: Yup.string()
      .required("Confirm Password is required")
      .oneOf(
        [Yup.ref("password")],
        "Password and Confirm Password must match."
      ),
  });

  const formOptions = { resolver: yupResolver(validationSchema) };

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm(formOptions);

  const onSubmit = (data) => {
    alert("Registered");
    navigate("/");
  };

  return (
    <div className="container">
      <form action="post" onSubmit={handleSubmit(onSubmit)}>
        <div className="left-side">
          <div className="heading">
            <h1>Sign Up</h1>
            <p>We do not share your personal details with anyone.</p>
          </div>
        </div>

        <div className="right-side">
          <div className="field">
            <input
              {...register("firstname", {
                required: "First Name is required",
                minLength: 1,
                maxLength: 20,
                pattern: /^[A-Za-z]+$/i,
              })}
              type="text"
              id="firstname"
              required
            />
            <label htmlFor="firstname">First Name</label>
            <p>{errors.firstname?.message}</p>
          </div>

          <div className="field">
            <input
              {...register("lastname", {
                required: "Last Name is required",
                minLength: 1,
                maxLength: 20,
                pattern: /^[A-Za-z]+$/i,
              })}
              type="text"
              id="lastname"
              required
            />
            <label htmlFor="lastname">Last Name</label>
            <p>{errors.lastname?.message}</p>
          </div>

          <div className="field">
            <input
              {...register("email", {
                required: "Email is required",
              })}
              type="email"
              id="email"
              required
            />
            <label htmlFor="email">Email</label>
            <p>{errors.email?.message}</p>
          </div>

          <div className="field">
            <input
              {...register("password", {
                required: "Password is required",
                minLength: 6,
              })}
              type="password"
              id="password"
              autoComplete="off"
              required
            />
            <label htmlFor="password">Password</label>

            <p>{errors.password?.message}</p>
          </div>

          <div className="field">
            <input
              {...register("confirmpassword", {
                required: "Password and Confirm Password must match.",
              })}
              type="password"
              id="confirmpassword"
              autoComplete="off"
              required
            />
            <label htmlFor="confirmpassword">Confirm Password</label>
            <p>{errors.confirmpassword?.message}</p>
          </div>

          <Button className="button" type="submit">
            Signup
          </Button>
        </div>
      </form>
    </div>
  );
}

export default Register;
