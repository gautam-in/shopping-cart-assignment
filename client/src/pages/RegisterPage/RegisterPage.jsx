import React, { useEffect, useState } from "react";
import classes from "./RegisterPage.module.scss";
import { useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";
import { userCred } from "../../Redux/cred";
import { useDispatch, useSelector } from "react-redux";
export default function RegisterPage() {
  const history = useHistory();
  const dispatch = useDispatch();
  let presentEmail = useSelector((state) => state.userList.userCredentials);
  const [loading, setLoading] = useState(false);
  const {
    register,
    formState: { errors },
    handleSubmit,
    getValues,
  } = useForm();

  const onSubmission = (data) => {
    setLoading(true);
    const storeCred = presentEmail.find(
      (user) => user.email === data.email.trim()
    );

    if (storeCred) {
      alert("You are already registered with this email kindly login ");
      history.push("/signin");
    } else {
      dispatch(
        userCred({
          id: new Date().getMilliseconds(),
          email: data.email.trim(),
          password: data.password.trim(),
        })
      );
      alert("Congrats ! You are Registered");
      setLoading(false);

      history.push("/signin");
    }
  };

  useEffect(() => {
    document.title = "S B | Register";
  }, []);

  return (
    <div className={classes.Container}>
      <section className={classes.PrimaryContainer}>
        <h1>Signup</h1>
        <p>We do not share your personal details with anyone.</p>
      </section>
      <section className={classes.SecondaryContainer}>
        <form
          className={classes.Form}
          autoComplete="off"
          onSubmit={handleSubmit(onSubmission)}
        >
          <fieldset
            style={{ border: "none" }}
            disabled={loading}
            aria-busy={loading}
          >
            <div className={classes.InputContainer}>
              <input
                className={classes.Input}
                id="fname"
                {...register("fname")}
                type="text"
                required
                autoComplete="off"
                placeholder="First Name"
              />
              <label className={classes.Label} htmlFor="fname">
                First Name
              </label>

              <p>{errors.fname?.message}</p>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="lname"
                className={classes.Input}
                {...register("lname")}
                type="text"
                autoComplete="off"
                required
                placeholder="Last Name"
              />
              <label className={classes.Label} htmlFor="lname">
                Last Name
              </label>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="email"
                className={classes.Input}
                type="email"
                {...register("email")}
                required
                autoComplete="off"
                placeholder="Email"
              />

              <label className={classes.Label} htmlFor="email">
                Email
              </label>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="password"
                className={classes.Input}
                autoComplete="off"
                type="password"
                {...register("password")}
                required
                placeholder="Password"
              />
              <label className={classes.Label} htmlFor="password">
                Password
              </label>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="passwordConfirmation"
                className={classes.Input}
                autoComplete="off"
                type="password"
                {...register("passwordConfirmation", {
                  required: "Please confirm password!",
                  validate: {
                    matchesPreviousPassword: (value) => {
                      const { password } = getValues();
                      return password === value || "Passwords should match!";
                    },
                  },
                })}
                required
                placeholder="Confirm Password"
              />
              <label className={classes.Label} htmlFor="Repassword">
                Confirm Password
              </label>
              {errors.passwordConfirmation && (
                <p style={{ color: "red" }}>
                  {errors.passwordConfirmation.message}
                </p>
              )}
            </div>
            <div className={classes.InputContainer}>
              <button type="submit" className={classes.Button}>
                Signup
              </button>
            </div>
          </fieldset>
        </form>
      </section>
    </div>
  );
}
