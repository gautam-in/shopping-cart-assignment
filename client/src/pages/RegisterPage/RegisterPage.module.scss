.Container {
    display: flex;
    width: 100%;

}

.ErrorContainer {
    position: absolute;

}

.PrimaryContainer {
    margin-left: 18%;

    h1 {
        font-weight: bolder;
        font-size: 2.3rem;
        margin: 0;
    }

    p {
        font-weight: 800;
        min-width: 370px;
    }
}

.SecondaryContainer {
    width: 50%;
    margin-left: 1.2rem;

}

.Form {
    display: flex;
    flex-direction: column;
    justify-content: center;

}

.InputContainer {
    position: relative;
    width: 50%;
    margin-bottom: 1rem;
}

.Input {
    width: 100%;
    min-width: 300px;
    height: 5vh;
    border: none;
    outline: none;
    border-bottom: 2px solid #c1c1c1;
    padding: 10px 0px 5px;
    transition: all 0.3s;
    color: #636262;

    &::placeholder {
        color: transparent;
    }
}

.Label {
    position: absolute;
    left: 0px;
    transition: all 0.3s;
    top: 50%;
    transform: translateY(-30%);
    cursor: text;
    color: #b4b4b4;
    font-weight: 500;
}

.Input:focus+Label,
.Input:not(:placeholder-shown)+Label {
    top: 0;
    left: 0;
    transform: translateY(-30%);
    color: #8fd7dc;
    font-size: 85.7%;
    cursor: default;
}

.Input:focus,
.Input:not(:placeholder-shown) {
    border-bottom: 2px solid #8fd7dc;
}

.Button,
.Button:visited {
    background: #c33761;

    border: none;
    color: white;
    font-size: 1.3rem;
    padding: 0.3rem;
    width: 100%;
    height: 5vh;
    cursor: pointer;
}

.Button:hover,
.Button:active,
.Button:focus {
    background: #df1350;

}

@media (max-width: 961px) {
    .Container {
        flex-direction: column;
        justify-content: center;
        align-items: center;
        margin-left: 10%;
    }

    .PrimaryContainer {
        margin-left: 0;
    }

    .SecondaryContainer {
        width: 62%;
    }

    .InputContainer {
        width: 100%;
    }
}

@media (max-width: 600px) {
    .SecondaryContainer {
        width: 100%;
    }

    .InputContainer {
        width: 100%;
    }

    .Container {
        margin-left: 0;
    }

    .PrimaryContainer {
        margin-left: 30%;
    }
}