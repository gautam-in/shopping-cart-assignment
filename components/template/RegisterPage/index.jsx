import { Component } from "react";
import { Row, Col, Button } from 'reactstrap';
import PageLayout from "../../organism/PageLayout";
import RegisterForm from "../../organism/RegisterForm";
class RegisterPage extends Component {
    render(){
       return (
          <PageLayout>
             <Row>
                <Col md={6} className="pt-3 pl-5">
                   <h2>Signup</h2>
                   <p>We do not share your personal details with anyone.</p>
                </Col>
                <Col md={6}>
                   <RegisterForm />
                </Col>
             </Row>
          </PageLayout>
       )
    }
}

export default RegisterPage;