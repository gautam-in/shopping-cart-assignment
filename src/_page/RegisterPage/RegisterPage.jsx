import React from 'react';
import { connect } from 'react-redux';
import { userActions } from '../../_actions';
import Footer from '../../_components/footer/footer';
import Header from '../../_components/header/header';
import '../LoginPage/style.scss';


class RegisterPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                firstName: '',
                lastName: '',
                email: '',
                password: '',
                confirm_password:''
            },
            errors: {},
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const { name, value } = event.target;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                [name]: value
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;
        if (this.validate()) {
            this.props.register(user);
        }
    }

    validate(){
        let user = this.state.user;
        let errors = {};
        let isValid = true;

        if (!user["firstName"]) {
            isValid = false;
            errors["firstName"] = "Please enter First Name.";
         }

         if (!user["lastName"]) {
            isValid = false;
            errors["lastName"] = "Please enter Last Name.";
         }
         
        if (typeof user["email"] !== "undefined") {
          var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
          if (!pattern.test(user["email"])) {
            isValid = false;
            errors["email"] = "Please enter valid email address.";
          }
        }

        if (!user["password"]) {
            isValid = false;
            errors["password"] = "Please enter password.";
         }

        if (typeof user["password"] !== "undefined") {
            
            var pattern = new RegExp(/^(?=.*[a-z])(?=.*[0-9])(?=.{7,})/i);
            if (!pattern.test(user["password"])) {
              isValid = false;
              errors["password"] = "Please enter valid password.";
            }
          }
          if (!user["confirm_password"]) {
            isValid = false;
            errors["confirm_password"] = "Please enter confirm password.";
          }
          if (typeof user["password"] !== "undefined" && typeof user["confirm_password"] !== "undefined") {
            if (user["password"] != user["confirm_password"]) {
              isValid = false;
              errors["confirm_password"] = "Passwords don't match.";
            }
        }
        this.setState({
          errors: errors
        });
    
        return isValid;
    }
    render() {
        const { user } = this.state;
        return (
        <div>
           <Header />
            <div className="main">

            <div className="title">

                <h2>SignUp</h2>
                <h5>We don't share personal data with anyone.</h5>
            </div>
            <div className="regForm">

            <form onSubmit={this.handleSubmit}>
            <div >
                <input type="text" name="firstName" placeholder=" " value={user.firstName} onChange={this.handleChange} />
                <label htmlFor="firstName">First Name</label>
                <p >{this.state.errors.firstName}</p>
            </div>
            <div>
                <input type="text" className="form-control" placeholder=" " name="lastName" value={user.lastName} onChange={this.handleChange} />
                <label htmlFor="lastName">Last Name</label>
                <p >{this.state.errors.lastName}</p>
            </div>
            <div>
                <input type="text" className="form-control" placeholder=" " name="email" value={user.email} onChange={this.handleChange} />
                <label htmlFor="email">Email</label>
                <p >{this.state.errors.email}</p>
            </div>
            <div>
                <input type="password" className="form-control" placeholder=" " name="password" value={user.password} onChange={this.handleChange} />
                <label htmlFor="password">Password</label>
                <p >{this.state.errors.password}</p>
                
            </div>
            <div>
                <input type="password" name="confirm_password" placeholder=" " value={user.confirm_password} onChange={this.handleChange} className="form-control" id="confirm_password" />
                <label htmlFor="confirm_password">Confirm Password</label>
                <p >{this.state.errors.confirm_password}</p>
                
            </div>
            <div className="submit">
                <input type="submit" name="" value="SignUp"/>
            </div>
            </form>

            </div>

        </div>
        <div className='regFooter'>
            <Footer />
        </div>
    </div>
        );
    }
}

function mapState(state) {
    const { registering } = state.registration;
    return { registering };
}

const actionCreators = {
    register: userActions.register
}

const connectedRegisterPage = connect(mapState, actionCreators)(RegisterPage);
export { connectedRegisterPage as RegisterPage };