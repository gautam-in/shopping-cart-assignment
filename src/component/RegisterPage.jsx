import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import { selectUser } from "../store/slices/user/user.selector";
import Button from "./Button";
import FormInput from "./FormInput";
import "../styles/register.scss";
import { signUpwithEmail } from "../store/slices/user/user.action";

const RegisterPage = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const user = useSelector(selectUser);

  const defaultFormFields = {
    firstName: "",
    lastName: "",
    email: "",
    password: "",
    confirmPassword: "",
  };

  const [formFields, setFormFields] = useState(defaultFormFields);
  const { firstName, lastName, email, password, confirmPassword } = formFields;

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (password !== confirmPassword) {
      alert("password do not match");
      return;
    }
    try {
      dispatch(signUpwithEmail(email, password));
      // const user = await createAuthUserWithEmailAndPassword(email, password);
      user && navigate("/");
      setFormFields(defaultFormFields);
    } catch (error) {
      if (error.code === "auth/email-already-in-use") alert("user exists");
      console.log("user creation failed", error.message);
    }
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormFields({
      ...formFields,
      [name]: value,
    });
  };

  return (
    <form onSubmit={handleSubmit} className="sign-up-container">
      <div className="sign-up-header">
        <h2 className="title">SignUp</h2>
        <span>We do not share your details with anyone.</span>
      </div>
      <div>
        <FormInput
          type="text"
          name="firstName"
          value={firstName}
          label="First Name"
          required
          onChange={handleChange}
        />
        <FormInput
          type="text"
          name="lastName"
          value={lastName}
          label="Last Name"
          required
          onChange={handleChange}
        />
        <FormInput
          type="email"
          name="email"
          value={email}
          label="Email"
          required
          onChange={handleChange}
        />
        <FormInput
          type="password"
          name="password"
          value={password}
          label="Password"
          required
          onChange={handleChange}
          inputMode="numeric"
          pattern="^(?=.*[a-zA-Z])(?=.*[0-9])(?!.* ).{6,}$"
        />
        <FormInput
          type="password"
          name="confirmPassword"
          value={confirmPassword}
          label="Confirm Password"
          required
          onChange={handleChange}
        />
        <div className="buttons-container">
          <Button type="submit" buttonType="max" onSubmit={handleSubmit}>
            Sign In
          </Button>
        </div>
      </div>
    </form>
  );
};

export default RegisterPage;
