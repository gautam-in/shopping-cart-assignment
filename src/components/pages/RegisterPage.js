import React from 'react';
import formInputs from '../../shared/Register.json';
import * as Constants from '../../shared/constants';
import FormPage from '../templates/FormPage/FormPage';
import { useHistory } from 'react-router-dom';
const RegisterPage = () => {
  const history = useHistory();
  const registerUser = () => {
    history.push('/home');
  };
  return (
    <main className='container'>
      <FormPage
        title={Constants.RegisterTitle}
        desc={Constants.RegisterDescription}
        formInputs={formInputs}
        formSubmit={registerUser}
      />
    </main>
  );
};

export default RegisterPage;
