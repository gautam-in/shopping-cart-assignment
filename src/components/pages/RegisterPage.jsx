import React, { useState, useEffect } from 'react'
import RegisterForm from "../Molecules/RegisterForm";

const RegisterPage = () => {
    useEffect(() => {
      document.title = 'Signup - Sabka Bazar'
  }, []);
    return (
        <>
        <div className="container top-container">
          <div className="row">
              <div className="col-md-5 pd-3">
                <h1>Signup</h1>
                <br/>
                <p>
                    We do not share your personal details with anyone.
                </p>
              </div>
              <div className="col-md-5 pd-3">
                <RegisterForm />
              </div>
          </div>
        </div>
        </>
    );
}

export default RegisterPage;