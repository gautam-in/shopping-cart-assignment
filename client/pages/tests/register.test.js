import React from "react";
import { render } from "@testing-library/react";
import RegisterPage from "../register.page";

describe("RegisterPage", () => {
  it("should render the component correctly", () => {
    const RegisterPageComponent = render(<RegisterPage />);
    expect(RegisterPageComponent).toMatchSnapshot();
  });
});
