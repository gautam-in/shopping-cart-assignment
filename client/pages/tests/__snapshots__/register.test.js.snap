// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage should render the component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="SectionWrapperstyles__SectionWrapper-sc-1qizg9n-0 gZaGKd"
      >
        <div
          class="RegistrationFormstyles__RegistrationFormWrapper-sc-ns6y4i-0 eEGcQx"
        >
          <section
            class="RegistrationFormstyles__RegistrationFormHeading-sc-ns6y4i-1 eNRGWB"
          >
            <h1>
              Signup
            </h1>
            <p>
              We do not share your personal details with anyone.
            </p>
          </section>
          <form
            class="RegistrationFormstyles__Form-sc-ns6y4i-2"
          >
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="firstName"
                name="firstName"
                title="First name"
                type="text"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="firstName"
              >
                First name
              </label>
            </div>
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="lastName"
                name="lastName"
                title="Last name"
                type="text"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="lastName"
              >
                Last name
              </label>
            </div>
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="email"
                name="email"
                title="Email"
                type="email"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="email"
              >
                Email
              </label>
            </div>
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="password"
                name="password"
                title="Password"
                type="password"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="password"
              >
                Password
              </label>
            </div>
            <div
              class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
            >
              <input
                class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
                id="confirmPassword"
                name="confirmPassword"
                title="Confirm password"
                type="password"
              />
              <label
                class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
                for="confirmPassword"
              >
                Confirm password
              </label>
            </div>
            <button
              aria-disabled="true"
              class="Buttonstyles__SubmitButton-sc-5pqk3u-0 exksWu"
              disabled=""
              id="submit"
              name="Signup"
              tabindex="0"
              title="Signup"
              type="submit"
            >
              Signup
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="SectionWrapperstyles__SectionWrapper-sc-1qizg9n-0 gZaGKd"
    >
      <div
        class="RegistrationFormstyles__RegistrationFormWrapper-sc-ns6y4i-0 eEGcQx"
      >
        <section
          class="RegistrationFormstyles__RegistrationFormHeading-sc-ns6y4i-1 eNRGWB"
        >
          <h1>
            Signup
          </h1>
          <p>
            We do not share your personal details with anyone.
          </p>
        </section>
        <form
          class="RegistrationFormstyles__Form-sc-ns6y4i-2"
        >
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="firstName"
              name="firstName"
              title="First name"
              type="text"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="firstName"
            >
              First name
            </label>
          </div>
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="lastName"
              name="lastName"
              title="Last name"
              type="text"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="lastName"
            >
              Last name
            </label>
          </div>
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="email"
              name="email"
              title="Email"
              type="email"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="email"
            >
              Email
            </label>
          </div>
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="password"
              name="password"
              title="Password"
              type="password"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="password"
            >
              Password
            </label>
          </div>
          <div
            class="TextFieldstyles__TextFieldWrapper-sc-t64aen-0 kQAkYf"
          >
            <input
              class="TextFieldstyles__Input-sc-t64aen-1 hwdbyY"
              id="confirmPassword"
              name="confirmPassword"
              title="Confirm password"
              type="password"
            />
            <label
              class="TextFieldstyles__Label-sc-t64aen-2 etFUBF"
              for="confirmPassword"
            >
              Confirm password
            </label>
          </div>
          <button
            aria-disabled="true"
            class="Buttonstyles__SubmitButton-sc-5pqk3u-0 exksWu"
            disabled=""
            id="submit"
            name="Signup"
            tabindex="0"
            title="Signup"
            type="submit"
          >
            Signup
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
